stringer.go

```diff
@@ -150,7 +164,7 @@ func compositeNode(seen map[node]bool, n node) bool {
 		}
 		return false
 
-	case *reference, *literal, *parseable:
+	case *reference, *literal, *parseable, *negation:
```

Comment:
Negation should be its own case and recurse into its sub-expression. See eg. `*strct`.